app/bundles/EmailBundle/Entity/Email.php

```diff
@@ -1124,8 +1124,8 @@ private function decodeAmpersands(&$content)
             foreach ($matches[0] as $url) {
                 $newUrl = $url;
 
-                while (false !== strpos($newUrl, '&amp;')) {
-                    $newUrl = str_replace('&amp;', '&', $newUrl);
+                while (false !== strpos($newUrl, '&amp;') || false !== strpos($newUrl, '&#38;') || false !== strpos($newUrl, '&#x26;')) {
+                    $newUrl = str_replace(['&amp;', '&#38;', '&#x26;'], '&', $newUrl);
                 }
```

Comment:
I can see that you found this decoding to be done in multiple places and fixed it everywhere. That's great! How about to make the code a bit DRY-er and encapsulate it with a static method in the [UrlHelper](https://github.com/mautic/mautic/blob/5.x/app/bundles/CoreBundle/Helper/UrlHelper.php)?

This way it will be fairly easy to write unit tests for it.

Reply:
Better! I created the static method in the UrlHelper.